test/gtest/logcumsumexp.hpp

```diff
@@ -0,0 +1,345 @@
+    }
+
+    void Verify()
+    {
+        // Computation error of fp16 is ~2^13 (=8192) bigger than
+        // the one of fp32 because mantissa is shorter by 13 bits.
+        double tolerance = std::is_same<T, float>::value ? 1.5e-6 : 8.2e-3;
+
+        // bf16 mantissa has 7 bits, by 3 bits shorter than fp16.
+        if(std::is_same<T, bfloat16>::value)
+            tolerance *= 8.0;
+
+        auto error_output = miopen::rms_range(ref_output, output);
+        ASSERT_EQ(miopen::range_distance(ref_output), miopen::range_distance(output));
+        EXPECT_LT(error_output, tolerance)
+            << "Error forward Output beyond tolerance Error: " << error_output
+            << " Tolerance: " << tolerance;
+    }
+
+    LogCumSumExpTestCase logcumsumexp_config;
+
+    tensor<T> input;
+    tensor<T> output;
+
+    tensor<T> ref_output;
+
+    miopen::Allocator::ManageDataPtr input_dev;
+    miopen::Allocator::ManageDataPtr output_dev;
+};
+
+template <typename T>
+struct LogCumSumExpTestBwd : public ::testing::TestWithParam<LogCumSumExpTestCase>
+{
+protected:
+    void SetUp() override
+    {
+        auto&& handle        = get_handle();
+        logcumsumexp_config  = GetParam();
+        auto gen_value_input = [](auto...) {
+            return prng::gen_descreet_uniform_sign<T>(1e-2, 100);
+        };
+        auto gen_value_doutput = [](auto...) {
+            return prng::gen_descreet_uniform_sign<T>(1e-2, 100);
+        };
+
+        auto lengths = logcumsumexp_config.lengths;
+
+        auto input_strides = GetStrides(lengths, logcumsumexp_config.contiguous);
+        input              = tensor<T>{lengths, input_strides}.generate(gen_value_input);
+        dinput             = tensor<T>{lengths, input_strides};
+
+        output  = tensor<T>{lengths};
+        doutput = tensor<T>{lengths}.generate(gen_value_doutput);
+
+        // Calculate output tensor value by forwarding input tensor
+        cpu_logcumsumexp_forward(input,
+                                 output,
+                                 logcumsumexp_config.dim,
+                                 logcumsumexp_config.exclusive,
+                                 logcumsumexp_config.reverse);
+
+        ref_dinput = tensor<T>{lengths, input_strides};
+
+        input_dev   = handle.Write(input.data);
+        output_dev  = handle.Write(output.data);
+        doutput_dev = handle.Write(doutput.data);
+        dinput_dev  = handle.Write(dinput.data);
+    }
+
+    void RunTest()
+    {
+        cpu_logcumsumexp_backward<T>(input,
+                                     output,
+                                     doutput,
+                                     ref_dinput,
+                                     logcumsumexp_config.dim,
+                                     logcumsumexp_config.exclusive,
+                                     logcumsumexp_config.reverse);
+
+        auto&& handle = get_handle();
+        auto status   = miopen::LogCumSumExpBackward(handle,
```

Comment:
Changed this due to packing LogCumSumExpBackward into namespace logcumsumexp

Reply:
Fixed